rfcs/0003-build-contents-diff.md

```diff
@@ -0,0 +1,152 @@
+- CONFIG is the same
+- BUILDPACK is different because it prints the Bill of Materials which represents the results of the buildpacks used instead of only the buildpack/s that caused the new build to kick off
+- STACK is the same now, but will be different once we have more stack metadata such as the dpackage list (and someday a CVE list) that we can print
+ 
+Competitor examples: 
+https://github.com/GoogleContainerTools/container-diff
+
+## Complexity/Risks
+- Would this be helpful for diagnosing failed builds or builds that don't start? If so, how can we surface this info without a BoM being generated? 
+- How does all this show up in CI, if at all? 
+
+## Mockups 
+- Lines preceded by + or - will be bolded and colored green or red respectively 
+- Added a table as a header to the diff output that gives context on name, date/time, and tags for the two images (in case you diff builds from two different images)
+- CONFIG CHANGES includes the 5 lines above and below the diff
+- BILL OF MATERIALS (BUILDPACKS) CHANGES prints only info on the buildpacks that have changed 
+- STACK CHANGES includes a diff of mix-ins if they change version names
+ 
+------------------
+``` 
+$kp build diff petclinic index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+
+IMAGE NAME        BUILD TIME                 LATEST IMAGE 
+petclinic         2020-08-04 10:12:46        index.docker.io/username/instance@sha256:afdsniuubadsiugbn3784htw70835qioresgkdjfvba3493w8uiebgd
+petclinic         2020-09-08 12:08:36        index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+
+------------------------------
+
+COMMIT CHANGES
+- Revision: 234890hpe9vjsr8gse98rghser
++ Revision: 43t789wghges87h540eq8378ge
+ 
+CONFIG CHANGES
+  env:
+  - name: "JAVA_BP_ENV"
+    value: "value"
+  resources:
+    requests:
+-     cpu: "0.25"
++     cpu: "0.5"
+      memory: "128M"
+    limits:
+      cpu: "0.5"
+      memory: "256M"
+ 
+BILL OF MATERIALS (BUILDPACKS) CHANGES
+local:
+- buildpack:
+    id: org.cloudfoundry.openjdk
+    version: 1.0.0-RC03
+  metadata:
+    licenses:
+    - type: GPL-2.0 WITH Classpath-exception-2.0
+      uri: https://openjdk.java.net/legal/gplv2+ce.html
+    name: OpenJDK JDK
+    sha256: 90c33cf3f2ed0bd773f648815de7347e69cfbb3416ef3bf41616ab1c4aa0f5a8
+    stacks:
+    - io.buildpacks.stacks.bionic
+    - org.cloudfoundry.stacks.cflinuxfs3
+    uri: https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.4%!B(MISSING)11/OpenJDK11U-jdk_x64_linux_hotspot_11.0.4_11.tar.gz
+  name: openjdk-jdk
+-  version: 11.0.4
++  version: 11.0.5
+
+STACK CHANGES
+- RunImage: index.docker.io/paketobuildpacks/run@sha256:34978twgsui4789et0wh9384erhsg0e798riuhgs0er7iugshe98riugeh99rgaser892
++ RunImage: index.docker.io/paketobuildpacks/run@sha256:fh34578tg29wb645rwhs87e5hrw0e98jtfnws9e7w378qe49tr3g39n5gwe9u5hge549w
+
+Mixins:
+- libgts-0.7-4
++ libgts-0.7-5
+```
+
```

Comment:
Could we add a processes section? This matches the output of `pack inspect-image` and is useful information if there are any changes.